### ISSUE TITLE
Error while building module 'std' imported from input_line_1:1:...

### ISSUE THREAD

**Issue by AAAlvesJr** [root-project/root#13450]
### Check duplicate issues.

- [ ] Checked for duplicates

### Description

Cant build ROOT on Fedora 38: 
```
[ 78%] Completed 'clad'
[ 78%] Built target clad
[ 78%] Generating G__Core.cxx, ../lib/Core.pcm
[ 78%] Building CXX object core/metacling/src/CMakeFiles/MetaCling.dir/TClingTypeInfo.cxx.o
While building module 'Core':
While building module 'std' imported from input_line_1:1:
In file included from <module-includes>:52:
In file included from /usr/include/c++/13/memory_resource:60:
In file included from /usr/include/c++/13/bits/memory_resource.h:46:
/usr/include/c++/13/bits/utility.h:219:27: error: redefinition of '__is_in_place_type_v<in_place_type_t<_Tp>>'
    inline constexpr bool __is_in_place_type_v<in_place_type_t<_Tp>> = true;
                          ^
/usr/include/c++/13/bits/stl_pair.h:62:11: note: '/usr/include/c++/13/bits/utility.h' included multiple times, additional include site in header from module 'std.bits/stl_pair.h'
# include <bits/utility.h> // for std::tuple_element, std::tuple_size
          ^
/opt/root_build/etc/cling/std.modulemap:505:10: note: std.bits/stl_pair.h defined here
  module "bits/stl_pair.h" {
         ^
/usr/include/c++/13/bits/memory_resource.h:46:11: note: '/usr/include/c++/13/bits/utility.h' included multiple times, additional include site in header from module 'std.memory_resource'
# include <bits/utility.h>              // index_sequence
          ^
/opt/root_build/etc/cling/std.modulemap:257:10: note: std.memory_resource defined here
  module "memory_resource" [optional] {
         ^
input_line_1:1:10: fatal error: could not build module 'std'
#include <new>
 ~~~~~~~~^
Warning in cling::IncrementalParser::CheckABICompatibility():
  Failed to extract C++ standard library version.
While building module 'Core':
While building module 'Cling_Runtime' imported from input_line_2:1:
While building module 'Cling_Runtime_Extra' imported from /opt/root_build/etc/cling/Interpreter/RuntimeUniverse.h:27:
In file included from <module-includes>:1:
/opt/root_build/etc/cling/Interpreter/DynamicExprInfo.h:13:10: fatal error: could not build module 'std'
#include <string>
 ~~~~~~~~^
/opt/root_build/etc/cling/Interpreter/RuntimeUniverse.h:27:10: fatal error: could not build module 'Cling_Runtime_Extra'
#include "cling/Interpreter/Visibility.h"
 ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from input_line_3:1:
In file included from /opt/root_build/etc/cling/lib/clang/13.0.0/include/assert.h:8:
In file included from /usr/include/assert.h:35:
In file included from /usr/include/features.h:515:
/usr/include/gnu/stubs.h:7:11: fatal error: 'gnu/stubs-32.h' file not found
# include <gnu/stubs-32.h>
          ^~~~~~~~~~~~~~~~
While building module 'Core':
While building module 'ROOT_Rtypes' imported from /opt/root_build/include/Rtypes.h:23:
In file included from <module-includes>:1:
/opt/root_build/include/RtypesCore.h:28:10: fatal error: could not build module 'std'
#include <cstddef> // size_t, NULL
 ~~~~~~~~^
/opt/root_build/include/ThreadLocalStorage.h:60:10: fatal error: could not build module 'ROOT_Rtypes'
#include "RtypesCore.h"
 ~~~~~~~~^~~~~~~~~~~~~~
While building module 'Core':
While building module 'ROOT_Foundation_Stage1_NoRTTI' imported from /opt/root_build/include/Rtypes.h:195:
In file included from <module-includes>:2:
/opt/root_build/include/ROOT/RStringView.hxx:25:10: fatal error: could not build module 'std'
#include <string_view>
 ~~~~~~~~^
Error: Error loading the default rootcling header files.
gmake[2]: *** [core/CMakeFiles/G__Core.dir/build.make:476: core/G__Core.cxx] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:31578: core/CMakeFiles/G__Core.dir/all] Error 2
gmake[1]: ** Esperando que outros processos terminem.
[ 78%] Building CXX object core/metacling/src/CMakeFiles/MetaCling.dir/TClingValue.cxx.o
[ 78%] Built target MetaCling
gmake: *** [Makefile:156: all] Error 2
```

### Reproducer

```
# The latest stable branch gets updated automatically on each release.
# You may update your local copy by issuing a `git pull` command from within `root_src/`.
$ git clone --branch latest-stable --depth=1 https://github.com/root-project/root.git root_src
$ mkdir root_build root_install && cd root_build
$ cmake -DCMAKE_INSTALL_PREFIX=../root_install ../root_src # && check cmake configuration output for warnings or errors
$ cmake --build . -- install -j4 # if you have 4 cores available for compilation
$ source ../root_install/bin/thisroot.sh # or thisroot.{fish,csh}
```

### ROOT version

Latest from github : `$ git clone --branch latest-stable --depth=1 https://github.com/root-project/root.git root_src`

### Installation method

building from source 

### Operating system

linux fedora 38

**Comment by jalopezg-git** [comment]
Hi @AAAlvesJr, thanks for reporting!  We do, however, have a Fedora 38 build as part of the CI...  Which compiler version are you using?

**Comment by jalopezg-git** [comment]
Related post in the forum that I mentioned yesterday: https://root-forum.cern.ch/t/root-6-28-04-installation-problem-from-source-on-fedora-38/56083/2

**Comment by hahnjo** [comment]
So it's been quite some time, but I already backported a fix in commit 9e1fd1d91d / https://github.com/root-project/root/pull/12901, it's just not released yet as 6.28/06. @Axel-Naumann
